File: JS/51-60/e_51.js
def controlMayus(frase):
    for letra in frase:
        if letra.isupper():
            return True
    return False

# Ejemplo de uso:
frase1 = "Esta es una Frase de Ejemplo"
frase2 = "esta es una frase en minúsculas"

print(controlMayus(frase1))  # Devolverá True
print(controlMayus(frase2))  # Devolverá False


// Esta función itera sobre cada caracter en la frase y verifica si es una letra mayúscula usando el método isupper().
// Si encuentra al menos una letra mayúscula, devuelve True; de lo contrario, devuelve False.

File: Java/src/e51/e_51.java
package e51;

public class e_51 {
    public static boolean controlMayus(String frase) {
        for (int i = 0; i < frase.length(); i++) {
            if (Character.isUpperCase(frase.charAt(i))) {
                return true;
            }
        }
        return false;
    }

    public static void main(String[] args) {
        String frase1 = "Esta es una Frase de Ejemplo";
        String frase2 = "esta es una frase en minúsculas";

        System.out.println(controlMayus(frase1)); // Devolverá true
        System.out.println(controlMayus(frase2)); // Devolverá false
    }
}

// Esta función controlMayus itera sobre cada carácter en la frase y utiliza el
// método Character.isUpperCase() para verificar si el carácter en la posición
// actual es una letra mayúscula. Si encuentra al menos un carácter en
// mayúscula, devuelve true; de lo contrario, devuelve false.